pkg/ccl/changefeedccl/show_changefeed_jobs_test.go

```diff
@@ -116,6 +119,39 @@ func TestShowChangefeedJobsBasic(t *testing.T) {
 	cdcTest(t, testFn, feedTestOmitSinks("webhook", "sinkless"), feedTestNoExternalConnection)
 }
 
+func TestShowChangefeedJobsShowsHighWaterTimestamp(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	testFn := func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
+		sqlDB := sqlutils.MakeSQLRunner(s.DB)
+		sqlDB.Exec(t, `CREATE TABLE foo(a INT PRIMARY KEY)`)
+		foo := feed(t, f, `CREATE CHANGEFEED FOR foo WITH resolved='0s',min_checkpoint_frequency='0s'`)
+
+		defer closeFeed(t, foo)
+
+		var highWaterTimestamp gosql.NullFloat64
+		var readableHighWaterTimestamp gosql.NullTime
+
+		// Wait for the high water timestamp to be non-null.
+		testutils.SucceedsSoon(t, func() error {
+			sqlDB.QueryRow(t, "SELECT high_water_timestamp, readable_high_water_timestamp from [SHOW CHANGEFEED JOBS]").Scan(&highWaterTimestamp, &readableHighWaterTimestamp)
+
+			if !highWaterTimestamp.Valid {
+				return errors.Errorf("high water timestamp not populated: %v", highWaterTimestamp)
+			}
+
+			return nil
+		})
+
+		// Timestamps in CockroachDB have microsecond precision by default
+		highWaterTimestampTime := time.Unix(0, int64(highWaterTimestamp.Float64)).UTC().Round(time.Microsecond)
+		require.Equal(t, highWaterTimestampTime, readableHighWaterTimestamp.Time)
+	}
+
+	cdcTest(t, testFn)
```

Comment:
as Miles said, you'll need to add `feedTestOmitSinks("sinkless")` back otherwise this test will fail whenever the sinkless feed is picked.

you can verify this to be the case by running the test with the reverse option `feedTestForceSink("sinkless")`

Reply:
Another way to test (often a good idea for tests that use `cdcTest` supporting multiple sinks) is to stress it (default count is 1k): `./dev test pkg/ccl/changefeedccl -f=TestShowChangefeedJobsShowsHighWaterTimestamp --stress --count=200`
<!-- Reviewable comment -OCJqV929Mmq7XZKmWeo:b-7jfg5a -->
<!-- Sent from Reviewable.io -->